rcl/include/rcl/event.h

```diff
@@ -47,7 +47,7 @@ typedef enum rcl_subscription_event_type_t
 } rcl_subscription_event_type_t;
 
 /// rmw struct.
-typedef struct rmw_event_t rmw_event_t;
+typedef struct rmw_event_s rmw_event_t;
```

Comment:
I know you didn't add this code, but this line actually feels wrong.  We should never be redefining `rmw_*` structures in this file, just using them.

Elsewhere in this package, we have lines like:

```
typedef rmw_serialized_message_t rcl_serialized_message_t;
```

I *suspect* that was what was originally intended, but it got crossed up somehow.

Regardless, I think we should either remove this line completely (and just use `rmw_event_t` everywhere), or change this to:

```
typedef rmw_event_t rcl_event_t;
```

And change the function signatures to match.  Thoughts?

Reply:
I can only guess, but I *think* it was intentional. `rcl/event.h` does not include `rmw/event.h`, so this would be a forward declaration. Of course, we can drop this and simply include the corresponding header.

Reply:
> Of course, we can drop this and simply include the corresponding header.

Yeah, that seems preferable to me.

Reply:
See https://github.com/ros2/rcl/pull/933/commits/af75126a45ffd600dd62d8c61746dc5f9e16e65b.